src/ng/components/dnd/drop.js

```diff
@@ -0,0 +1,88 @@
+import Directive from '../directive';
+import EventListener from 'core/infrastructure/event.listener';
+import {GRID_NAME, DROP_NAME, CAN_DROP_NAME, DROP_EFFECT_NAME, ON_DROP_NAME} from 'src/definition';
+
+class Drop extends Directive(DROP_NAME) {
+	constructor($scope, $element) {
+		super();
+
+		this.$scope = $scope;
+		this.element = $element[0];
+		this.listener = new EventListener(this, this.element);
+	}
+
+	onInit() {
+		this.element.classList.add('can-drop');
+		this.listener.on('dragenter', this.enter);
```

Comment:
no look at this.listener = new EventListener(THIS, element)
this is sepcial abstraction to symplify life =)